Checkstyle: Fix line length violations in Matches

ssoloff opened the pull request:
This PR fixes a single violation of the Checkstyle LineLength rule in the `Matches` class.

RoiEXLab commented on src/main/java/games/strategy/triplea/delegate/BattleDelegate.java:
```diff
@@ -571,8 +571,10 @@ private static void setupTerritoriesAbandonedToTheEnemy(final BattleTracker batt
+    final Iterator<Territory> battleTerritories = Match
+        .getMatches(
+            data.getMap().getTerritories(),
+            Matches.territoryHasEnemyUnitsThatCanCaptureItAndIsOwnedByTheirEnemyAndIsNotUnownedWater(player, data))
```
This is way too long of a method name!
As this has only one usage: What about splitting this into multiple matches and grouping them together with `Match#allOf`?

ssoloff replied:
There's no way I'm going to attempt to refactor this method without having characterization tests in place. :smile:

We really don't have any precedent for writing tests for `Match` implementations, which is probably not a good thing.  I've been meaning for a while to see how hard it would be to write some `Match` tests.  So maybe this would be a good opportunity to do that.

I'll take a stab at it, but if it starts getting extremely painful, I'm just going to punt and say the bang-for-buck value is too low to justify to fix a single Checkstyle violation.

ssoloff commented:
@RoiEXLab I split the method in question into two simpler methods and combined them into a composite match with `allOf()`, as you suggested.  The commits reflect the actions I performed:

1. Write characterization tests for the original method.
1. Split the original method into two methods with their own individual tests derived from the characterization tests.
1. Replace the implementation of the original method with a composite of the two new methods, while ensuring the original characterization tests continue to pass.
1. Remove the original method, and replace its lone caller with a composite of the two new methods.

Since I now had tests in place, I also refactored the implementation a bit, so it may look slightly different than you would expect after an Extract Method.

Also, I discovered a check that can never be true and removed it.  `Territory#getOwner()` should never return `null` because `Territory#setOwner()` maps a `null` argument to `PlayerID#NULL_PLAYERID`.  So I simply removed this check.

I left the individual commits intact to make it easier for you to review.  Please feel free to squash before merging.

ssoloff commented:
I added some new tests because I was missing coverage on one of the branches in `territoryIsNotUnownedWater()`.  (I don't understand why it's important for an unowned territory to not have a `TerritoryAttachment`, so I figure it's just safer to cover all the conditions as added by the original author.)
